neo_batterylevelshutdown/usb.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import os
+import subprocess
+from distutils.dir_util import copy_tree
+
+
+
+class USB:
+
+    def __init__(self):
+        pass
+
+    def isUsbPresent(self, devPath='/dev/sda1'):
+        '''
+        Returns if there is a USB plugged into specified devPath
+        :return: True / False
+        '''
+        logging.debug("Checking to see if usb is mounted")
+        return os.path.exists(devPath)
+
+
+    def unmount(self, curPath='/media/usb0'):
+        '''
+        Unmount the USB drive from curPath
+        :return:  True / False
+        '''
+        try:
+            logging.debug("Unmounting file at location {}".format(curPath))
+            response = subprocess.call(['umount', curPath])  # unmount drive
+            return True if response == 0 else False
+        except:
+            return False
+
+    def mount(self, devPath='/dev/sda1', newPath='/media/usb1'):
+        '''
+        Mount the USB drive at the devPath to the specified newPath location
+
+        :return: True / False
+        '''
+
+        # try:
+        logging.debug("Mounting USB at {} to {}".format(devPath, newPath))
+        if not os.path.exists(newPath):  # see if desired mounting directory exists
+            os.makedirs(newPath)  # create directory and all of the intermediary directories
+        response = subprocess.call(['mount', '-o', 'sync,noexec,nodev,noatime,nodiratime,utf8',
+                                    devPath, newPath])
+        logging.debug("Response: {}".format(response))
+        return True if response == 0 else False
+        # except:
+        #     return False
+
+    def copyFiles(self, sourcePath='/media/usb1', destPath='/media/usb0'):
+        '''
+        Move files from sourcePath to destPath recursively
+        :param sourcePath: place where files are
+        :param destPath:  where we want to copy them to
+        :return:  True / False
+        '''
+
+        if os.path.exists(sourcePath) and os.path.exists(destPath):
+            logging.debug("Copying tree")
+            try:
+                copy_tree(sourcePath, destPath)
+                logging.debug("Done copying")
+                return True
+            except:
```

Comment:
As above - please catch the most specific exception possible.

Reply:
This has been addressed